rmf_traffic/include/rmf_traffic/Trajectory.hpp

```diff
@@ -43,270 +42,115 @@ class Trajectory
     /// times of all subsequent Trajectory segments, use adjust_finish_times()
     /// instead.
     ///
-    /// \warning If you change the finishing time value of this Segment such
-    /// that it falls directly on another Segment's finish time, you will get a
+    /// \warning If you change the finishing time value of this Waypoint such
+    /// that it falls directly on another Waypoint's finish time, you will get a
     /// std::invalid_argument exception, because discontinuous jumps are not
     /// supported, and indicate a significant mishandling of trajectory data,
     /// which is most likely a serious bug that should be remedied.
     ///
-    /// \note If this Segment's finish time crosses over another Segment's
+    /// \note If this Waypoint's finish time crosses over another Waypoint's
     /// finish time, that signficantly changes the topology of the Trajectory,
     /// because it will change the order in which the positions are passed
     /// through.
     ///
     /// \param[in] new_time
-    ///   The new finishing time for this Trajectory Segment.
+    ///   The new finishing time for this Trajectory Waypoint.
     ///
     /// \sa adjust_finish_times(Time new_time)
-    Segment& set_finish_time(Time new_time);
+    Waypoint& change_time(Time new_time);
 
-    /// Adjust the finishing time of this segment and all subsequent segments by
+    /// Adjust the finishing time of this waypoint and all subsequent segments by
     /// the given duration. This is guaranteed to maintain the ordering of the
-    /// Trajectory Segments, and is more efficient than changing all the times
+    /// Trajectory Waypoints, and is more efficient than changing all the times
     /// directly.
     ///
     /// \warning If a negative delta_t is given, it must not cause this
-    /// Segment's finish time to be less than or equal to the finish time of its
-    /// preceding Segment, or else a std::invalid_argument exception will be
+    /// Waypoint's finish time to be less than or equal to the finish time of its
+    /// preceding Waypoint, or else a std::invalid_argument exception will be
     /// thrown.
     ///
     /// \param[in] delta_t
-    ///   How much to change the finishing time of this segment and all later
+    ///   How much to change the finishing time of this waypoint and all later
     ///   segments. If negative, it must not cross over the finish time of the
-    ///   previous segment, or else a std::invalid_argument will be thrown.
+    ///   previous waypoint, or else a std::invalid_argument will be thrown.
     ///
     /// \sa set_finish_time(Time new_time)
-    void adjust_finish_times(Duration delta_t);
-
-    /// Compute the motion across this Trajectory segment
-    std::unique_ptr<Motion> compute_motion() const;
+    void adjust_times(Duration delta_t);
 
     class Implementation;
   private:
 
     /// \internal Private constructor. Use Trajectory::add_segment() to create
```

Comment:
Does the API contain this function? If so perhaps it should also be renamed
```suggestion
    /// \internal Private constructor. Use Trajectory::add_waypoint() to create
```

Reply:
Ah that's a very very outdated comment. The function should be `Trajectory::insert(~)` now.